iceoryx_hoofs/testing/include/iceoryx_hoofs/testing/error_reporting/test_support.hpp

```diff
@@ -0,0 +1,158 @@
+#include "iox/static_lifetime_guard.hpp"
+
+#include <thread>
+#include <utility>
+
+// NOLINTNEXTLINE(hicpp-deprecated-headers) required to work on some platforms
+#include <setjmp.h>
+
+namespace iox
+{
+namespace testing
+{
+
+using ErrorHandler = iox::StaticLifetimeGuard<iox::testing::TestErrorHandler>;
+
+/// @brief indicates whether the test error handler registered a specific error
+template <typename Code>
+inline bool hasError(Code&& code)
+{
+    auto e = iox::err::toError(std::forward<Code>(code));
+    return ErrorHandler::instance().hasError(e.code(), e.module());
+}
+
+/// @brief indicates whether the test error handler invoked panic
+bool hasPanicked();
+
+/// @brief indicates whether the test error handler registered any error
+bool hasError();
+
+/// @brief indicates whether the test error handler registered a precondition violation
+bool hasPreconditionViolation();
+
+/// @brief indicates whether the test error handler registered an assumption violation
+bool hasAssumptionViolation();
+
+/// @brief indicates whether the test error handler registered  violation (there are only two kinds).
+bool hasViolation();
+
+/// @brief indicates there is no error, violation or panic.
+bool isInNormalState();
+
+/// @brief runs testFunction in a testContext that can detect fatal failures;
+/// runs in the same thread
+/// @note uses setjmp/longjmp
+template <typename Function, typename... Args>
+inline void testContext(Function&& testFunction, Args&&... args)
+{
+    jmp_buf* buf = ErrorHandler::instance().prepareJump();
+
+    if (buf == nullptr)
+    {
+        // should not fail with correct usage
+        GTEST_FAIL();
+    };
+
+    // setjmp must be called in a stackframe that still exists when longjmp is called
+    // Therefore there cannot be a convenient abstraction that does not also
+    // know the test function that is being called.
+    // NOLINTNEXTLINE(cert-err52-cpp) exception cannot be used, required for testing to jump in case of failure
+    if (setjmp(&(*buf)[0]) != ErrorHandler::instance().jumpIndicator())
+    {
+        testFunction(std::forward<Args>(args)...);
+    }
+}
+
+/// @brief runs testFunction in a testContext that can detect fatal failures;
+/// runs in a separate thread
+/// @note uses a longjump inside the thread it runs the function in
+template <typename Function, typename... Args>
+inline void runInTestThread(Function&& testFunction, Args&&... args)
+{
+    // needed to infer the testContext arguments
+    auto f = [&]() { testContext(std::forward<Function>(testFunction), std::forward<Args>(args)...); };
+
+    std::thread t(f);
+    if (t.joinable())
+    {
+        t.join();
+    }
+}
+
+} // namespace testing
+} // namespace iox
+
+// Use macros to preserve line numbers in tests (failure case).
+
+// ASSERT_* aborts test if the check fails.
+
+// NOLINTBEGIN(cppcoreguidelines-macro-usage) macro required for source location in tests
+
+#define ASSERT_OK() ASSERT_TRUE(iox::testing::isInNormalState())
```

Comment:
All of these macros need to be prefixed with at least `IOX_`. Even better would be `IOX_TESTING_` to not confuse the user with the `IOX_ASSUME` macro.

Reply:
Done